### PR TITLE
csi: Update gRPC errors

### LINKED ISSUE #242: All CSI grpc errors must be checked to statisfy new CSI spec

### PR DESCRIPTION
**Pull request by lpabon**
Closes #242

Signed-off-by: Luis Pabón <luis@portworx.com>

### CODE REVIEW

File: csi/controller.go [hunk 1]
```diff
@@ -340,18 +348,30 @@ func (s *OsdCsiServer) CreateVolume(
...
+		// Check the requested arguments match that of the existing volume
+		if v.GetSpec().GetSize() != req.GetCapacityRange().GetRequiredBytes() {
+			return nil, status.Errorf(
+				codes.AlreadyExists,
+				"Existing volume has a size of %v which differs from requested size of %v",
+				v.GetSpec().GetSize(),
+				req.GetCapacityRange().GetRequiredBytes())
+		}
+		if v.GetSpec().GetShared() != csiRequestsSharedVolume(req) {
+			msg := "Existing volume is not a shared volume while request is asking for a shared volume"
```
[the remark this answers is not in the document]
Comment (lpabon) [reply csi/controller.go]: It is not an assumption, it is initializing the variable with that value. The next line will change it if it is shared. It is just as if I had said:

[the remark this answers is not in the document]
Comment (piyush-nimbalkar) [reply csi/controller.go]: I don't see the shared being initialized to a certain value. We get the existing volume whose shared value could be either true or false, right? The condition is not equivalent to !shared as you mentioned in  the example. It is v.shared != csi_request.shared

[the remark this answers is not in the document]
Comment (lpabon) [reply csi/controller.go]: Got it. Update coming.

**Comment by piyush-nimbalkar** [comment]
LGTM

**Pull request merged**